v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java

```diff
@@ -176,31 +235,66 @@ static PipelineResult executeShardedMigration(
 
     List<String> tablesToMigrate =
         PipelineController.listTablesToMigrate(options.getTables(), schemaMapper, ddl);
+    Set<String> tablesToMigrateSet = new HashSet<>(tablesToMigrate);
```

Comment:
Bunch of code seems duplicated - is it possible to refactor into a single method ?

Reply:
Skipping as per offline discussion